Firmware/Board/v3/board.cpp

```diff
@@ -0,0 +1,416 @@
+        encoders[0], // encoder
+        sensorless_estimators[0], // sensorless_estimator
+        controllers[0], // controller
+        fet_thermistors[0], // fet_thermistor
+        motor_thermistors[0], // motor_thermistor
+        motors[0], // motor
+        trap[0], // trap
+        endstops[0], endstops[1], // min_endstop, max_endstop
+    },
+    {
+        1, // axis_num
+#if HW_VERSION_MAJOR == 3 && HW_VERSION_MINOR >= 5
+        7, // step_gpio_pin
+        8, // dir_gpio_pin
+#else
+        3, // step_gpio_pin
+        4, // dir_gpio_pin
+#endif
+        osPriorityHigh, // thread_priority
+        encoders[1], // encoder
+        sensorless_estimators[1], // sensorless_estimator
+        controllers[1], // controller
+        fet_thermistors[1], // fet_thermistor
+        motor_thermistors[1], // motor_thermistor
+        motors[1], // motor
+        trap[1], // trap
+        endstops[2], endstops[3], // min_endstop, max_endstop
+    },
+}};
+
+
+
+#if (HW_VERSION_MINOR == 1) || (HW_VERSION_MINOR == 2)
+Stm32Gpio gpios[] = {
+    {nullptr, 0}, // dummy GPIO0 so that PCB labels and software numbers match
+
+    {GPIOB, GPIO_PIN_2},
+    {GPIOA, GPIO_PIN_5},
+    {GPIOA, GPIO_PIN_4},
+    {GPIOA, GPIO_PIN_3},
+    {nullptr, 0},
+    {nullptr, 0},
+    {nullptr, 0},
+    {nullptr, 0},
+
+    {GPIOB, GPIO_PIN_4}, // ENC0_A
+    {GPIOB, GPIO_PIN_5}, // ENC0_B
+    {GPIOA, GPIO_PIN_15}, // ENC0_Z
+    {GPIOB, GPIO_PIN_6}, // ENC1_A
+    {GPIOB, GPIO_PIN_7}, // ENC1_B
+    {GPIOB, GPIO_PIN_3}, // ENC1_Z
+    {GPIOB, GPIO_PIN_8}, // CAN_R
+    {GPIOB, GPIO_PIN_9}, // CAN_D
+};
+#elif (HW_VERSION_MINOR == 3) || (HW_VERSION_MINOR == 4)
+Stm32Gpio gpios[] = {
+    {nullptr, 0}, // dummy GPIO0 so that PCB labels and software numbers match
+
+    {GPIOA, GPIO_PIN_0},
+    {GPIOA, GPIO_PIN_1},
+    {GPIOA, GPIO_PIN_2},
+    {GPIOA, GPIO_PIN_3},
+    {GPIOB, GPIO_PIN_2},
+    {nullptr, 0},
+    {nullptr, 0},
+    {nullptr, 0},
+
+    {GPIOB, GPIO_PIN_4}, // ENC0_A
+    {GPIOB, GPIO_PIN_5}, // ENC0_B
+    {GPIOA, GPIO_PIN_15}, // ENC0_Z
+    {GPIOB, GPIO_PIN_6}, // ENC1_A
+    {GPIOB, GPIO_PIN_7}, // ENC1_B
+    {GPIOB, GPIO_PIN_3}, // ENC1_Z
+    {GPIOB, GPIO_PIN_8}, // CAN_R
+    {GPIOB, GPIO_PIN_9}, // CAN_D
+};
+#elif (HW_VERSION_MINOR == 5) || (HW_VERSION_MINOR == 6)
+Stm32Gpio gpios[GPIO_COUNT] = {
+    {nullptr, 0}, // dummy GPIO0 so that PCB labels and software numbers match
+
+    {GPIOA, GPIO_PIN_0},
+    {GPIOA, GPIO_PIN_1},
+    {GPIOA, GPIO_PIN_2},
+    {GPIOA, GPIO_PIN_3},
+    {GPIOC, GPIO_PIN_4},
+    {GPIOB, GPIO_PIN_2},
+    {GPIOA, GPIO_PIN_15},
+    {GPIOB, GPIO_PIN_3},
+    
+    {GPIOB, GPIO_PIN_4}, // ENC0_A
+    {GPIOB, GPIO_PIN_5}, // ENC0_B
+    {GPIOC, GPIO_PIN_9}, // ENC0_Z
+    {GPIOB, GPIO_PIN_6}, // ENC1_A
+    {GPIOB, GPIO_PIN_7}, // ENC1_B
+    {GPIOC, GPIO_PIN_15}, // ENC1_Z
+    {GPIOB, GPIO_PIN_8}, // CAN_R
+    {GPIOB, GPIO_PIN_9}, // CAN_D
+};
+#else
+#error "unknown GPIOs"
+#endif
```

Comment:
Please add a comment next to each one so it's easy for human to see which gpio is what at a glance.